mm2src/mm2_main/src/lp_swap/taker_swap.rs

```diff
@@ -837,6 +845,24 @@ impl TakerSwap {
         }
     }
 
+    async fn get_taker_fee_data(&self) -> Result<SwapTxDataMsg, MmError<PaymentInstructionsErr>> {
+        // If taker fee is a lightning payment the payment hash will be sent in the message
+        let taker_fee_data = self
+            .r()
+            .taker_fee
+            .as_ref()
+            .expect("TakerSwapMut::taker_fee must be some value to use get_taker_fee_data")
```

Comment:
Could you please add `# Panic` section at the method doc comments, or just return an error here instead of unwrapping?